orchagent/pfcactionhandler.h

```diff
@@ -39,15 +47,28 @@ class PfcWdActionHandler
         }
 
         static void initWdCounters(shared_ptr<Table> countersTable, const string &queueIdStr);
+        void initCounters(void);
+        void commitCounters(void);
+
+        virtual bool getHwCounters(PfcWdHwStats& counters)
```

Comment:
>getHwCounters [](start = 21, length = 13)

What is Hw counter? Is there any Sw counter?

Reply:
HW counters are actual counters from ASIC, They can be either queue/PG counters or ACL counters (depending on handler). PFC WD counters calculate an overall traffic that was transmitted or dropped during PFC storm based on HW counters values.